src/auth/guards/roles.guard.ts

```diff
@@ -0,0 +1,21 @@
+import { Injectable, CanActivate, ExecutionContext } from '@nestjs/common';
+import { Reflector } from '@nestjs/core';
+import { ADMIN_KEY } from '../decorators/adminonly.decorator';
+
+@Injectable()
+export class RolesGuard implements CanActivate {
+  constructor(private reflector: Reflector) {}
+
+  canActivate(context: ExecutionContext): boolean {
+    const requiredAdminRole = this.reflector.getAllAndOverride<boolean>(ADMIN_KEY, [
+      context.getHandler(),
+      context.getClass(),
+    ]);
+
+    if (!requiredAdminRole) {
+      return true;
+    }
+    const { user } = context.switchToHttp().getRequest();
+    return user.isAdmin;
+  }
+}
```

Comment:
Add trailing newline to file

```suggestion
}

```